src/components/ZoomableImage.vue

```diff
@@ -0,0 +1,90 @@
+<script setup>
+import { defineProps, ref} from 'vue'
+// eslint-disable-next-line no-unused-vars
+const props = defineProps(['imageSrc'])
+const image = ref(null)
+const imageLeftPos = ref(0)
+const imageTopPos = ref(0)
+const currentZoom = ref(1)
+
+const MIN_ZOOM = 1
+const MAX_ZOOM = 3
+const STEP_SIZE = 0.1
+let mouseX
+let mouseY
+
+const zoom = (e, direction) => {
+	if (!direction) direction = e.deltaY > 0 ? -1 : 1
+	let newZoom = currentZoom.value + direction * STEP_SIZE 
+	if (newZoom > MIN_ZOOM && newZoom < MAX_ZOOM) currentZoom.value = newZoom
+}
+
+// save the position clicked and add the pointermove listener
+const pointerDown = (e) => {
+	mouseX = e.pageX
+	mouseY = e.pageY
+	image.value.onpointermove = pointerMove
+}
+
+// use mouse movement delta to set position, limit so image stays on screen
+const pointerMove = (e) => {
+	let x 						= imageLeftPos.value+e.pageX-mouseX
+	let y 						= imageTopPos.value+e.pageY-mouseY
+	// TODO change this to be bounded by the condition of whether overflow will happen for a more natural use
+	if( x > -800 && x < 800) imageLeftPos.value   = x
+	if( y > -800 && y < 800) imageTopPos.value    = y
+	mouseX 						= e.pageX
+	mouseY 						= e.pageY
+}
+
+</script>
+<template>
+  <div
+    class="image_container"
+    :style="{height: image?.naturalHeight, width: image?.naturalWidth}"
+  >
+    <img
+      ref="image"
+      :src="imageSrc"
+      class="image"
+      :style="{left:imageLeftPos+'px', top:imageTopPos+'px', transform: `scale(${currentZoom})`}"
+      @wheel="zoom"
+      @pointerdown.prevent="pointerDown"
```

Comment:
when the mouse is released or the mouse leaves the image area we want to stop listening for move events aka stop moving the image around